typescript/cli/src/utils/input.ts

```diff
@@ -53,3 +71,393 @@ export async function inputWithInfo({
+
+          setOptionState({
+            options: organizeItems(
+              Object.values(updatedDataMap),
+              selectableOptionsSeparator,
+            ),
+            currentOptionState: updatedDataMap,
+          });
+          setSearchTerm('');
+        }
+      } else {
+        setSearchTerm(rl.line);
+      }
+    });
+
+    const message = theme.style.message(config.message);
+
+    let description;
+    const page = usePagination({
+      items: optionState.options,
+      active,
+      renderItem({ item, isActive }) {
+        if (Separator.isSeparator(item)) {
+          return ` ${item.separator}`;
+        }
+
+        if (item.disabled) {
+          const disabledLabel =
+            typeof item.disabled === 'string' ? item.disabled : '(disabled)';
+          return theme.style.disabledChoice(`${item.name} ${disabledLabel}`);
+        }
+
+        if (isActive) {
+          description = item.description;
+        }
+
+        const checkbox = item.checked
+          ? theme.icon.checked
+          : theme.icon.unchecked;
+        const color = isActive ? theme.style.highlight : (x: string) => x;
+        const cursor = isActive ? theme.icon.cursor : ' ';
+        return color(`${cursor}${checkbox} ${item.name}`);
+      },
+      pageSize,
+      loop,
+    });
+
+    if (status === 'done') {
+      const selection = optionState.options.filter(isChecked);
+      const answer = theme.style.answer(
+        theme.style.renderSelectedChoices(selection, optionState.options),
+      );
+
+      return `${prefix} ${message} ${answer}`;
+    }
+
+    let helpTipTop = '';
+    let helpTipBottom = '';
+    const defaultTopHelpTip =
+      instructions ??
+      `(Press ${theme.style.key('tab')} to select, and ${theme.style.key(
+        'enter',
+      )} to proceed`;
+    const defaultBottomHelpTip = `\n${theme.style.help(
+      '(Use arrow keys to reveal more choices)',
+    )}`;
+    if (theme.helpMode === 'always') {
+      helpTipTop = theme.style.helpTip(defaultTopHelpTip);
+      helpTipBottom =
+        optionState.options.length > pageSize ? defaultBottomHelpTip : '';
+      firstRender.current = false;
+    } else if (theme.helpMode === 'auto' && firstRender.current) {
+      helpTipTop = theme.style.helpTip(defaultTopHelpTip);
+      helpTipBottom =
+        optionState.options.length > pageSize ? defaultBottomHelpTip : '';
+      firstRender.current = false;
+    }
+
+    const choiceDescription = description
+      ? `\n${theme.style.description(description)}`
+      : ``;
+
+    let error = '';
+    if (errorMsg) {
+      error = `${theme.style.error(errorMsg)}`;
+    } else if (
+      optionState.options.length === 0 &&
+      searchTerm !== '' &&
+      status === 'idle'
+    ) {
+      error = theme.style.error('No results found');
+    }
+
+    return `${prefix} ${message}${helpTipTop} ${searchTerm}\n${page}${helpTipBottom}${choiceDescription}${error}${ansiEscapes.cursorHide}`;
```

Comment:
This function is very long. would it be possible to break it up into more digestible chunks?

Reply:
yep I'll try to split it!